tools/targets/PSOC6.py

```diff
@@ -133,5 +147,327 @@ def find_cm0_image(toolchain, resources, elf, hexf, hex_filename):
+        with open(sdk_path / Path(sb_config["policy_file"])) as p:
+            policy_str = p.read()
+            policy_file = json.loads(policy_str)
+            p.close()
+
+            firmware_list = policy_file["boot_upgrade"]["firmware"]
+
+        # collect firmware descriptions from policy for corresponding images
+        firmware_cyb_cm0p = firmware_list[0]
+
+        if "_M0_" in target:
+            firmware_spe_cm0p = firmware_list[1]
+            firmware_nspe_cm4 = firmware_list[2]
+
+            slots = check_slots_integrity(toolchain, fw_cyb=firmware_cyb_cm0p, fw_spe=firmware_spe_cm0p,
+                                          fw_nspe=firmware_nspe_cm4, target_data=processing_target)
+        else:
+            firmware_nspe_cm4 = firmware_list[2]
+            slots = check_slots_integrity(toolchain, fw_cyb=firmware_cyb_cm0p, fw_nspe=firmware_nspe_cm4,
+                                          target_data=processing_target)
+    else:
+        # consider single stage bootloading scheme
+        with open(sdk_path / Path(sb_config["policy_file"])) as p:
+            policy_str = p.read()
+            policy_file = json.loads(policy_str)
+            p.close()
+
+        firmware_list = policy_file["boot_upgrade"]["firmware"]
+        firmware_cyb_cm0p = firmware_list[0]
+        firmware_nspe_cm4 = firmware_list[1]
+        slots = check_slots_integrity(toolchain, fw_cyb=firmware_cyb_cm0p,
+                                      fw_nspe=firmware_nspe_cm4, target_data=processing_target)
+
+    target_sig_data = [{"img_data": sb_config["boot0"], "slot_data": slots[0],
+                        "key_file": sb_config["priv_key_file"], "sdk_path": sdk_path, "id": slots[2]}]
+    
+    if slots[1] is not None:        
+        target_sig_data.append({"img_data": sb_config["boot1"], "slot_data": slots[1],
+                                "key_file": sb_config["priv_key_file"], "sdk_path": sdk_path, "id": slots[2]})
+        # check if slot1 image sould be encrypted
+        try:                                
+            if slots[1]["encrypt"] is True:
+            
+                dev_pub_key = sdk_path / Path(sb_config["dev_pub_key_file"])
+                if not os.path.isfile(str(dev_pub_key)):
+                    toolchain.notify.debug("[PSOC6.sign_image] ERROR: Device public key file not found in " + str(dev_pub_key))
+                    raise Exception("imgtool finished execution with errors!")            
+
+                aes_key_file = sdk_path / Path(sb_config["aes_key_file"])
+                if not os.path.isfile(str(aes_key_file)):
+                    toolchain.notify.debug("[PSOC6.sign_image] ERROR: AES-128 key file not found in " + str(aes_key_file))
+                    raise Exception("imgtool finished execution with errors!")   
+
+                target_sig_data[1].update({"aes_key": sb_config["aes_key_file"], "dev_pub_key": sb_config["dev_pub_key_file"]})
+
+        except KeyError:
```

Comment:
What key error is this exception catching? This try except block contains too much code.